lib/modules/backupAndRestore/providers/GoogleDrive.js

```diff
@@ -28,9 +34,46 @@ export class GoogleDrive extends Provider {
 			await Permissions.request(['https://accounts.google.com/o/oauth2/v2/auth', 'https://content.googleapis.com/drive/v3/*']);
 		});
 
+		// Re-check email address cache, in case it got invalidated due to google account logout
+		if (!this.getCachedEmailAddress()) {
```

Comment:
It doesn't seem like this will ever recheck after `emailAdress` for the first time.

Reply:
Because the email address cache is invalidated here when a user logs out of their google account: https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/4440/files#diff-ad1ac56107331cb3882bedd9fd80ffb0R17

The user could get logged out of their google account for several reasons:
1. Manual google account log out.
2. Cookies cleared.
3. Google for Business accounts are automatically logged out after 30 days.

Furthermore, if after logout, the user chooses to log in with a different google account, then this would email address would need to be updated. Note that with the login hint, the user would be prompted by Google to log into the same account which was stored here in local storage, but it's up to the user to log into the same or different google account.